### PR TITLE
`consistent-function-scoping`: Check anonymous functions

### PR DESCRIPTION
**Pull request by fisker**
```js
				const doFoo = function() {
					const doBar = function(bar) {
						return bar;
					};
				};
```

Function `doBar` should be able to move to outer scope.

### CODE REVIEW

File: test/consistent-function-scoping.js [hunk 1]
```diff
@@ -356,6 +390,47 @@ ruleTester.run('consistent-function-scoping', rule, {
...
+		},
+		{
+			code: outdent`
+				const doFoo = function() {
+					const doBar = function(bar) {
+						return bar;
+					};
+				};
+			`,
+			errors: [createError('function')]
```
Comment (fisker) [inline test/consistent-function-scoping.js:411]: This function should has name `doBar`, https://github.com/mysticatea/eslint-utils/pull/14

Comment (sindresorhus) [reply test/consistent-function-scoping.js:437]: Is this PR blocked by that?

Comment (fisker) [reply test/consistent-function-scoping.js:437]: No, we can improve message later.

**Comment by sindresorhus** [comment]
Can you fix the merge conflict?

**Pull request merged**